src/index.js

```diff
@@ -82,17 +96,23 @@ function splitGuide({
     const allPromises = fileObjs.reduce((all, fileObj) => {
       return [...all, ...saveFinalAndWorkshop(fileObj)]
     }, [])
-    return Promise.all(allPromises)
+    return promiseChunker(allPromises, 100)
   }
 
   function saveFinalAndWorkshop({file, workshopContents, finalContents}) {
     const relativeDestination = path.relative(templatesDir, file)
     const workshopDestination = path.resolve(exercisesDir, relativeDestination)
     const finalDestination = path.resolve(exercisesFinalDir, relativeDestination)
     return [
-      workshopContents ? saveFile(workshopDestination, workshopContents) : null,
-      finalContents ? saveFile(finalDestination, finalContents) : null,
-    ].filter(Boolean) // filter out the files that weren't saved
+      workshopContents ? [saveFile, workshopDestination, workshopContents] : null,
+      finalContents ? [saveFile, finalDestination, finalContents] : null,
+    ].filter(p => !!p) // filter out the files that weren't saved
```

Comment:
No. I think that was an error when I updated my branch with the latest from your repo. I'll revert it.